[docs] Add tablespace and partitioned index content (#11959)
yugabyte/yugabyte-db: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md

Regarding colocation, indexes follow their table. If the table is colocated, its index is also colocated; if the table is not colocated, its index is also not colocated.

### Partitioned Indexes

> Creating an index on a partitioned table will automatically create a corresponding index for every partition in the default tablespace. Conversely, it is also possible to create an index on each partition individually. Prefer using the latter approach to creating partitioned indexes in the following cases:

Reviewer: ```suggestion
Creating an index on a partitioned table automatically creates a corresponding index for every partition in the default tablespace. It's also possible to create an index on each partition individually, which you should do in the following cases:
```

> * Parallel writes are expected while creating the index: This is because concurrent builds for indexes on partitioned tables are not supported. In this case, it is better to use concurrent builds to create indexes on each partition individually.

Reviewer: Blank lines around lists...
```suggestion

* Parallel writes are expected while creating the index, because concurrent builds for indexes on partitioned tables aren't supported. In this case, it's better to use concurrent builds to create indexes on each partition individually.
```

> * [Row-Level Geo-Partitioning](../../../../../explore/multi-region-deployments/row-level-geo-partitioning/) is being used: In this case, create the index separately on each partition to customize the tablespace in which each index should be created.

Reviewer: ```suggestion
* [Row-level geo-partitioning](../../../../../explore/multi-region-deployments/row-level-geo-partitioning/) is being used. In this case, create the index separately on each partition to customize the tablespace in which each index is created.
```

Specify a list of columns which will be included in the index as non-key columns.

### TABLESPACE clause

> Specify the name of the [tablespace](../../../../../explore/ysql-language-features/going-beyond-sql/tablespaces/) that describes the placement configuration for this index. By default, indexes are placed in `pg_default` tablespace, which spreads the tablets of the index evenly across the cluster.

Reviewer: ```suggestion
Specify the name of the [tablespace](../../../../../explore/ysql-language-features/going-beyond-sql/tablespaces/) that describes the placement configuration for this index. By default, indexes are placed in the `pg_default` tablespace, which spreads the tablets of the index evenly across the cluster.
```
